src/actions/BringMoveSwap.ts

```diff
@@ -0,0 +1,266 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+} from "../Types";
+import { runForEachEditor } from "../targetUtils";
+import update from "immutability-helper";
+import displayPendingEditDecorations from "../editDisplayUtils";
+import { computeChangedOffsets } from "../computeChangedOffsets";
+import { flatten, zip } from "lodash";
+import { Selection, TextEditorDecorationType, TextEditor } from "vscode";
+
+interface DecorationTypes {
+  sourceStyle: TextEditorDecorationType;
+  sourceLineStyle: TextEditorDecorationType;
+  destinationStyle: TextEditorDecorationType;
+  destinationLineStyle: TextEditorDecorationType;
+}
+
+interface Edit {
+  editor: TextEditor;
+  range: Selection;
+  newText: string;
+  targetsIndex: number;
+  originalSelection: TypedSelection;
+}
+
+interface ThatMarkEntry {
+  editor: TextEditor;
+  targetsIndex: number;
```

Comment:
ohhh this is how you preserve the highlight color in "swap".  Nice

Reply:
:)